File: app/src/main/java/com/mrswimmer/shift/domain/interactor/SettingsService.java
package com.mrswimmer.shift.domain.interactor;

import android.content.Context;
import android.content.SharedPreferences;

import com.google.firebase.database.DatabaseReference;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class SettingsService {
    SharedPreferences sharedPreferences;

    public SettingsService(SharedPreferences sharedPreferences) {
        this.sharedPreferences = sharedPreferences;
    }

    void setUserId(String id) {
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putString("id", id);
        editor.apply();
    }

    String getUserId() {
        return sharedPreferences.getString("id", "error");
    }
}

File: app/src/main/java/com/mrswimmer/shift/domain/interactor/SettingsService.kt
package com.mrswimmer.shift.domain.interactor

import android.content.SharedPreferences

class SettingsService(var sharedPreferences: SharedPreferences) {

    fun getEmail(): String {
        return sharedPreferences.getString("email", "error")
    }

    fun setEmail(email: String) {
        var editor: SharedPreferences.Editor = sharedPreferences.edit()
        editor.putString("email", email)
        editor.apply()
    }
}